compiler/src/iree/compiler/Dialect/Stream/Analysis/Partitioning/ReferencePartitioning.cpp

```diff
@@ -204,13 +254,13 @@ partitionStreamableOpsReference(IREE::Stream::PartitioningConfigAttr config,
     if (firstCandidateOrdinal != -1) {
       LLVM_DEBUG(llvm::dbgs() << "Moving to first candidate partition "
                               << firstCandidateOrdinal << " (continue)\n");
-      builders[firstCandidateOrdinal]->insert(&op);
-      opInfo.membership.set(firstCandidateOrdinal);
-      opInfo.hazards.reset(firstCandidateOrdinal);
+      builders[firstCandidateOrdinal]->insert(&op, opInfo);
       continue;
     }
 
     // Mark the op as having hazards against all other partitions.
```

Comment:
@benvanik, could you shed some light on what was your reasoning here when you wrote the algorithm?